core/server/data/validation/index.js

```diff
@@ -42,6 +43,27 @@ validator.extend('isTimezone', function isTimezone(str) {
     return moment.tz.zone(str) ? true : false;
 });
 
+validator.extend('isEmptyOrWebsite', function isEmptyOrWebsite(str) {
```

Comment:
I created a new method for the following reason: The `website` input on the user profile page is supposed to be _only_ a website, which clearly doesn't include query strings for example.

Because I'm not entirely sure, how the validation for other URL fields should be (does a query string matter, or not), I didn't touch the existing `isEmptyOrUrl` method.

The validator package with version 6.2.1 includes now a validation for `<` and `>` which hasn't been there before, so `validator.isURL()` would cover an input like `https://example.com/<script>bla</script>/` and return `false`, but to make sure, I built in more safety and return `false` if we found a query string but `isURL` validated the URL.